docs/intro/overview.rst

```diff
@@ -13,172 +13,83 @@ precisely, `web scraping`_), it can also be used to extract data using APIs
-``http://www.mininova.org/tor/NUMBER`` where ``NUMBER`` is an integer. We'll use
-that to construct the regular expression for the links to follow: ``/tor/\d+``.
-
-We'll use `XPath`_ for selecting the data to extract from the web page HTML
-source. Let's take one of those torrent pages:
-
-    http://www.mininova.org/tor/2676093
-
-And look at the page HTML source to construct the XPath to select the data we
-want which is: torrent name, description and size.
-
-.. highlight:: html
-
-By looking at the page HTML source we can see that the file name is contained
-inside a ``<h1>`` tag::
-
-   <h1>Darwin - The Evolution Of An Exhibition</h1>
-
-.. highlight:: none
-
-An XPath expression to extract the name could be::
-
-    //h1/text()
-
-.. highlight:: html
-
-And the description is contained inside a ``<div>`` tag with ``id="description"``::
-
-   <h2>Description:</h2>
-
-   <div id="description">
-   Short documentary made for Plymouth City Museum and Art Gallery regarding the setup of an exhibit about Charles Darwin in conjunction with the 200th anniversary of his birth.
-
-   ...
-
-.. highlight:: none
-
-An XPath expression to select the description could be::
 
-    //div[@id='description']
+    class StackOverflowSpider(scrapy.Spider):
+        name = 'stackoverflow'
+        start_urls = ['http://stackoverflow.com/questions?sort=votes']
 
-.. highlight:: html
+        def parse(self, response):
+            for href in response.css('.question-summary h3 a::attr(href)'):
+                full_url = response.urljoin(href.extract())
+                yield scrapy.Request(full_url, callback=self.parse_question)
 
-Finally, the file size is contained in the second ``<p>`` tag inside the ``<div>``
-tag with ``id=specifications``::
+        def parse_question(self, response):
+            title = response.css('h1 a::text').extract_first()
+            votes = response.css('.question .vote-count-post::text').extract_first()
+            tags = response.css('.question .post-tag::text').extract()
+            body = response.css('.question .post-text').extract_first()
+            yield {
+                'title': title,
+                'votes': votes,
+                'body': body,
+                'tags': tags,
+                'link': response.url,
+            }
 
-   <div id="specifications">
 
-   <p>
-   <strong>Category:</strong>
-   <a href="/cat/4">Movies</a> &gt; <a href="/sub/35">Documentary</a>
-   </p>
+Put this in a file, name it to something like ``stackoverflow_spider.py``
+and run the spider using the :command:`runspider` command::
 
-   <p>
-   <strong>Total size:</strong>
-   150.62&nbsp;megabyte</p>
+    scrapy runspider stackoverflow_spider.py -o top-stackoverflow-questions.json
 
 
-.. highlight:: none
+When this finishes you will have in the ``top-stackoverflow-questions.json`` file
+a list of the most upvoted questions in StackOverflow in JSON format, containing the
```

Comment:
Hmmm, I think I can work out something -- replacing the big fields so it won't turn out a mess.